pyresample/geometry.py

```diff
@@ -318,14 +324,49 @@ def get_bbox_lonlats(self, vertices_per_side: Optional[int] = None, force_clockw
             warnings.warn("The `frequency` argument is pending deprecation, use `vertices_per_side` instead",
                           PendingDeprecationWarning, stacklevel=2)
         vertices_per_side = vertices_per_side or frequency
-        lons, lats = self._get_bbox_elements(self.get_lonlats, vertices_per_side)
+        if self.is_geostationary:
+            lon_sides, lat_sides = self._get_geo_boundary_sides(vertices_per_side=vertices_per_side)
+        else: 
+            lon_sides, lat_sides = self._get_boundary_sides(self.get_lonlats, vertices_per_side)
         if force_clockwise and not self._corner_is_clockwise(
-                lons[0][-2], lats[0][-2], lons[0][-1], lats[0][-1], lons[1][1], lats[1][1]):
+                lon_sides[0][-2], lat_sides[0][-2],
+                lon_sides[0][-1], lat_sides[0][-1], 
+                lon_sides[1][1], lat_sides[1][1]):
             # going counter-clockwise
-            lons, lats = self._reverse_boundaries(lons, lats)
-        return lons, lats
+            lon_sides, lat_sides = self._reverse_boundaries(lon_sides, lat_sides)
+        return lon_sides, lat_sides
+    
 
-    def _get_bbox_elements(self, coord_fun, vertices_per_side: Optional[int] = None) -> tuple:
+    def _get_geo_boundary_sides(self, vertices_per_side=None):
+        """Retrieve the boundary sides list for geostationary projections."""
+        # Define default frequency
+        if vertices_per_side is None:
+            vertices_per_side = 50
+        # Ensure at least 4 points are used
+        if vertices_per_side < 4:
+            vertices_per_side = 4
+        # Ensure an even number of vertices for side creation
+        if (vertices_per_side % 2) != 0:
+            vertices_per_side = vertices_per_side + 1
+        lons, lats = _get_geostationary_bounding_box_in_lonlats(self, nb_points=vertices_per_side)
+        
+        # Retrieve dummy sides for GEO (side1 and side3 always of length 2)
+        # - TODO: _get_geostationary_bounding_box_in_lonlats now does not return nb_points ! 
+        side02_step = int(lons.shape[0] / 2) - 1 # 
```

Comment:
I added this line to fix the issue of having return less `nb_points` than `vertices_per_side` !